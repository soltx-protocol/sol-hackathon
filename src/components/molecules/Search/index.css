.search {
	display: flex;
	align-items: center;

	& > .title {
		margin-right: 10px;
		font-weight: 700;
		font-size: 16px;
		line-height: 24px;
		color: #d9dcff;
	}
}
