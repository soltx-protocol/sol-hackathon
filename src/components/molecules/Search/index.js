import React from 'react';
import classnames from 'classnames';

import InputSearch from 'components/atoms/InputSearch';

import PoolConfig from 'defi/configs.json';

import styles from './index.css';

const options = Object.keys(PoolConfig.pools)
	.filter(pool => PoolConfig.pools[pool].deprecated !== true)
	.map(pool => ({ ...PoolConfig.pools[pool], label: pool, value: pool }));

const Search = ({ className, onSearch, onClear }) => (
	<div className={classnames(styles.search, className)}>
		<div className={styles.title}>Pool Search</div>
		<InputSearch options={options} onSearch={onSearch} onClear={onClear} />
	</div>
);

export default Search;
