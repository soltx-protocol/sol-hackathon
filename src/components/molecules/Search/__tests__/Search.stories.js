import React from 'react';
import { storiesOf } from '@storybook/react';

import Search from 'components/molecules/Search';

const stories = storiesOf('molecules/Search', module);

stories.add('__interactive', () => <Search />);
