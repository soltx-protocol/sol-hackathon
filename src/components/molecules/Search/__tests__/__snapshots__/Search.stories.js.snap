// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <div
      class="title"
    >
      Pool Search
    </div>
    <div
      class="inputSearch"
    >
      <div
        class="content"
        role="button"
        tabindex="0"
      >
        <div
          class="input input icon isLeftIcon"
        >
          <input
            type="text"
            value=""
          />
          <button
            class="icon"
            type="button"
          >
            <test-file-stub />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
